lenny/core/api.py

```diff
@@ -51,13 +58,35 @@ def make_url(cls, path):
         if PORT and PORT not in {80, 443}:
             url += f":{PORT}"
         return f"{url}{path}"
-
+    
     @classmethod
-    def auth_check(cls, book_id: int, email=None):
-        # TODO: permission/auth checks go here
-        # for userid, store/check hashed email in db?
-        return True
+    def make_session_cookie(cls,email: str):
+        return cls.signer.sign(email).decode('utf-8')
     
+    @classmethod
+    def validate_session_cookie(cls, session_cookie: str):
+        """Validates the session cookie and returns the email if valid."""
+        if not session_cookie:
+            return None
+        try:
+            return cls.signer.unsign(session_cookie).decode('utf-8')
+        except BadSignature:
+            return None
+
+    @classmethod
+    def auth_check(cls, openlibrary_edition: int, email = None , session: str = None):
+        """
+        Checks if the user is allowed to access the book.
+        """
+        if item := Item.exists(openlibrary_edition):
+            if not item.is_login_required:
+                return item # open access book
+            session_email = cls.validate_session_cookie(session)
+            if not (email and session and email == session_email) and item.is_login_required:
+                return None # not authenticated user for open access book
+            return item
```

Comment:
```suggestion
            if session_email := cls.validate_session_cookie(session):
                return item if email and email == session_email else None
```

Reply:
Since the previous lines check `if not item.is_login_required:` we don't need to do that again here.

https://github.com/ArchiveLabs/lenny/pull/87/files#diff-daea9adfc6a724cd61b2c765b7ed1e9d21712d5bd3fce17212f4e66af731abcfR82